source/navitiacommon/navitiacommon/models.py

```diff
@@ -401,6 +401,19 @@ def get_from_id_or_name(cls, id=None, name=None):
         else:
             raise Exception({'error': 'instance is required'}, 400)
 
+    def delete_dataset(self):
+        jobs = db.session.query(Job).filter(Job.instance_id == self.id).all()
+        for j in jobs:
+            for dataset in j.data_sets.all():
+                if dataset.type == 'poi':
+                    db.session.query(Metric).filter(Metric.dataset_id == dataset.id).delete()
+                    db.session.delete(dataset)
+            db.session.commit()
+            # if no more datasets related to job, delete it
+            if not j.data_sets.all():
```

Comment:
does it re trigger a query every time?

Reply:
according to the doc:
all()
    Return the results represented by this Query as a list.
    This results in an execution of the underlying query.